views/application.py

```diff
@@ -93,3 +94,57 @@ def update(self, request, *args, **kwargs):
                 list(set(new_team_member_ids)-set(old_team_member_ids))
             )
         return response.Response(serializer.data)
+
+
+class ApplicationHiddenDetailView(generics.GenericAPIView):
+    """
+    View details of Application which should not be exposed directly
+    """
+
+    permission_classes = [permissions.IsAuthenticated, ]
+
+    def post(self, request, *args, **kwargs):
+        """
+        Authenticate if the requesting user is Team Member
+        of the application and check the password entered.
+        :param request: the request being processed
+        :param args: arguments
+        :param kwargs: keyword arguments
+        :return: response containing client_secret
+        """
+
+        try:
+            data = request.data
+            application_id = data.get('id')
+            user_password = data.get('password')
+            person = request.person
+            user = request.user
+            user_id = person.id
+            application = Application.objects.get(pk=application_id)
```

Comment:
You will need to handle a case where `id` is not present in the `POST` body or it is there but its type is wrong.
For example:
```json
{
  "id": "dfgf",
  "password": "sdfghjk"
}
```
```json
{
  "password": "sdfghjk"
}
```
All of the above examples will fail and throw an error `500`.

You might be wondering why will the frontend ever send `"sdfghjk"` as `id`. But recall, you can always draft custom requests using `cURL` or Postman.